server/server.js

```diff
@@ -0,0 +1,74 @@
+const users =  require('../server/users.js');
+const rooms = require('../server/rooms.js');
+const bookings = require('../server/bookings.js');
+const user = require('../server/rooms.js');
+
+const express = require('express');
+
+const app = express();
+
+app.set('port', process.env.port || 8080);
+
+app.use(express.json());
+
+app.locals.title = 'Overlook';
+
+app.locals.users = users;
+app.locals.rooms = rooms;
+app.locals.bookings = bookings;
+app.locals.user = user;
+
+app.get('/api/v1/users', (req, resp) => {
+  resp.status(200).json(app.locals.users);
+});
+
+app.get('/api/v1/users/:id', (req, resp) => {
+  const { id } = req.params;
+
+  const getUserById = app.locals.users.filter(user => user.id === +id)
+  resp.status(200).json(getUserById);
+})
+
+app.get('/api/v1/bookings', (req, resp) => {
+  resp.status(200).json(app.locals.bookings);
+});
+
+app.get('/api/v1/bookings/:id', (req, resp) => {
+  const { id } = req.params;
+
+  const getBookingById = app.locals.bookings.filter(booking => booking.id === +id)
+  resp.status(200).json(getBookingById);
+})
+
+app.post('/api/v1/bookings', (req, resp) => {
+  const requiredProp = ["userID", "date", "roomNumber"];
+  const recievedProp = Object.keys(req.body);
+
+  for (let prop of requiredProp) {
+    if (!recievedProp.includes(prop)) {
+      return resp.status(418).json({ error: `Your a little tea pot, add ${prop}` })
+    }
```

Comment:
Favorite error message ever. Although technically, this would be a "You're" situation.

Reply:
So cute!

Reply:
Per MDN:
The HTTP 418 I'm a teapot client error response code indicates that the server refuses to brew coffee because it is, permanently, a teapot. A combined coffee/tea pot that is temporarily out of coffee should instead return 503. This error is a reference to Hyper Text Coffee Pot Control Protocol defined in April Fools' jokes in 1998 and 2014.